File: cpp/test-catch2/example2a.cpp
#include "catch2/catch.hpp"
#include "example2a.h"


TEST_CASE ("ShoppingBasket") {
    SECTION("total over 100 gives five percent discount") {
        std::tuple<std::string, double> itemA = {"A", 10.0};
        std::tuple<std::string, double> itemB = {"B", 25.0};
        std::tuple<std::string, double> itemC = {"C", 9.99};
        auto allItems = std::vector<std::tuple<std::string, double>>();
        allItems.emplace_back(itemA);
        allItems.emplace_back(itemB);
        allItems.emplace_back(itemC);
        auto basket = std::vector<std::tuple<std::string, int>>();
        basket.emplace_back("A", 5);
        basket.emplace_back("B", 2);
        basket.emplace_back("C", 6);

        auto manager = InventoryManager(allItems);
        CHECK(6 == manager.getQuantity(basket, "C"));
        CHECK(151.94 == Approx(manager.calculateTotal(basket)).margin(0.01));
    }
}

File: python/example2a.py
class InventoryManager:
    pass


def get_quantity(all_items, basket, item_name):
    for itemN, item_quantity in basket:
        if itemN == item_name:
            return item_quantity
    return 0


def calculate_total(all_items, basket):
    all_items_dict = dict(all_items)
    total = 0
    for item_name, item_quantity in basket:
        total += all_items_dict[item_name] * item_quantity

    if total > 100:
        total = total * 0.95
    return total
